### ISSUE TITLE
Review got stuck in a loop

### ISSUE THREAD

**Issue by issekinichou** [brianrak/satori-reader-mobile-beta#68]
**Describe the bug**
When I got to the end of today's review card set on my iPad, message says 2 more cards (see screenshot). Tapped orange button to start another set. Temporary progress message getting fresh cards. Same page repeated (screenshot), 2 cards, orange button. Tap again, same loop.
 
**To Reproduce**
May be hard to reproduce. This is how I found it. Will update if I can add some cards and get them to become due.

Review: 44 cards due today. Finished set of 20. Exit, read article, added one vocab item. Finished final set. Message: 2 more cards (see screenshot).
Tapped orange button to start another set. 
Getting fresh cards. 
Same page repeated (screenshot), 2 cards, orange button. Tap again, same loop.

Tapped menu Articles to exit. Menu Review says 2 cards due. Tapped blue button. Loop repeated. Forced quit. Review ditto. Force sync ditto.
Tried web browser. Review 2 cards. Tapped button. Normal operation. The two cards were JE-EJ pair for a word I added today. I am fairly sure I added them on my iPad. I don't think I was using a web browser at that time. That might have caused a sync conflict.
Answered review first card easy, chose miss for second. 1 card due.
App force sync. 1 card due. Loop again.
Website review deleted new card.
App force sync. 0 cards due. Review all cards. NORMAL operation. Showed card 1 of 20. Answered first card and it proceeded as normal to second card. Exited to Articles.
 
**Expected behavior**
Tapping orange button should lead to normal review session.

**Screenshots**
![unadjustednonraw_thumb_2ecb](https://user-images.githubusercontent.com/42171096/45021490-a00aa500-b029-11e8-8c31-81438115ec95.jpg)

**Smartphone (please complete the following information):**
iPad mini 4
iOS 11.4.1
satori 1.0.2

**Additional context**

**Comment by issekinichou** [comment]
No problem today.
During review at 10/14, Added word from review full context article. X to return. numbers unchanged 10/14. Finish set, same message 2 cards due. Tap button, OK. Worked as normal. Answer 1 and choose miss for the other.
1 card due.
Tap button, again OK.

**Comment by issekinichou** [comment]
Encountered this loop bug again today. 
Hard to reproduce on demand, but I'll see if I can narrow down the conditions.

Review 40 cards (I think, didn't take a note) due today.
During review session, open full context, add a word. This time I added it during the first 20 set.
Finish 20. 24 due! Start another set. 
Shows 1/19! That is strange. I expected a full 20 with some to follow.
At 19/19 I had to take a break, exit to articles.
Return, Review. 6 cards due. Finish set. 1 card due. No misses in answers or new additions I think.
Tap orange button (1 card due) and get stuck in loop.
Switch to browser on website. Review: 1 card due. No problems. Finish set. O due.
Back to app. Menu Articles, Review, still shows 1 card due. Articles, Review still 1. Force sync. 0 due.

The key action seems to be adding a word (EJ-JE pair) from the full context overlay pane during a review session. The app gets messed up but the website is ok.

Not an urgent showstopper, but this needs fixing post-release. I hope you will get time to test it. I will try and take better notes when reviewing.

**Comment by issekinichou** [comment]
This bug appeared again today. It's definitely a real bug, but hard to reproduce on demand and hard to identify the required conditions.
Released app 1.04
iPad mini 4
iOS 11.4.1

I think I had 39 cards for review today.
Did 20 and took a break without exiting review.
Next set was 19 I think, but I didn't take notes.
After finishing, I card due. Tap button, stuck in loop.
Exit to Articles
Review: 1 card due. Tap blue button
1 card due, orange button.
Loop.
Website, Review 1 card due. Tap blue button. 1 of 1 亡くなる is one I added several days ago from Obon. Definitely not added today from Reviews' Reveal Full Context.
Answer numbers were 5/6/13. Clearly not a new card today. Answered 0 cards due
Back to app, waited for auto-sync. 0 cards due.
